packages/react/src/useUploadThing.ts

```diff
@@ -89,11 +85,17 @@ export const INTERNAL_uploadthingHookGen = <TRouter extends FileRouter>() => {
         opts?.onClientUploadComplete?.(res);
         return res;
       } catch (e) {
-        const error =
-          e instanceof UploadThingError ? e : fatalClientError(e as Error);
-        opts?.onUploadError?.(
-          error as UploadThingError<inferErrorShape<TRouter>>,
-        );
+        let error: UploadThingError<inferErrorShape<TRouter>>;
+        if (e instanceof UploadThingError) {
+          error = e as UploadThingError<inferErrorShape<TRouter>>;
+        } else {
+          error = INTERNAL_DO_NOT_USE__fatalClientError(e as Error);
+          console.error(
+            "Something went wrong. Please contact UploadThing and provide the following cause:",
+            error.cause,
```

Comment:
Does this potentially need to be stringified?

Reply:
can you not just log an error as is?

Reply:
You lose the `options.cause` if you just log it.
 
<img width="457" alt="image" src="https://github.com/pingdotgg/uploadthing/assets/11494384/22af762c-1159-406e-9ba5-557a0fa9dbb6">

Reply:
error.cause in our case should just be a normal error without extra props, but added a conditional toString() just in case